### PR TITLE
Fix #1192: Assignment starting with object literals

### LINKED ISSUE #1192: Assignment starting with object literal
$ coffee -e '{}.p = 0'

    .:2
      {}.p = 0;
        ^
    SyntaxError: Unexpected token .
    ...

    $ coffee -bpe '{}.p = 0; {}.p++'
    {}.p = 0;
    {}.p++;

Extreme edge cases, but still a bug.

### PR DESCRIPTION
**Pull request by lydell**

### CODE REVIEW

File: test/assignment.coffee [hunk 1]
```diff
@@ -131,6 +131,12 @@ test "more compound assignment", ->
   val ?= true
   eq c, val
 
+test "#1192: assignment starting with object literals", ->
+  doesNotThrow (-> CoffeeScript.run "{}.p = 0")
+  doesNotThrow (-> CoffeeScript.run "{}.p++")
+  doesNotThrow (-> CoffeeScript.run "{}[0] = 1")
```
Comment (michaelficarra) [inline test/assignment.coffee]: I'd like to also see `{a:{0:{}}}[0].a[0] = 0`

Comment (lydell) [inline test/assignment.coffee]: Added. Though I removed the first `[0]` assuming that’s what you meant. Correct me if I’m wrong.

Comment (michaelficarra) [inline test/assignment.coffee]: Yeah, thanks.
